src/test/java/com/android/volley/toolbox/DiskBasedAsyncCacheTest.java

```diff
@@ -0,0 +1,450 @@
+import static org.hamcrest.Matchers.emptyArray;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import androidx.annotation.Nullable;
+import com.android.volley.AsyncCache;
+import com.android.volley.Cache;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = 26)
+public class DiskBasedAsyncCacheTest {
+
+    private static final int MAX_SIZE = 1024 * 1024;
+
+    private DiskBasedAsyncCache cache;
+
+    private AsyncCache.OnCompleteCallback emptyCallback;
+
+    @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Rule public ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void setup() throws IOException {
+        emptyCallback =
+                new AsyncCache.OnCompleteCallback() {
+                    @Override
+                    public void onComplete() {
+                        // do nothing
+                    }
+                };
+        // Initialize empty cache
+        cache = new DiskBasedAsyncCache(temporaryFolder.getRoot(), MAX_SIZE);
+        cache.initialize(emptyCallback);
+    }
+
+    @After
+    public void teardown() {
+        cache = null;
+    }
+
+    @Test
+    public void testEmptyInitialize() throws ExecutionException, InterruptedException {
+        assertNull(getEntry("key").get());
+    }
+
+    @Test
+    public void testPutGetZeroBytes() throws ExecutionException, InterruptedException {
+        final Cache.Entry entry = new Cache.Entry();
+
+        entry.data = new byte[0];
+        entry.serverDate = 1234567L;
+        entry.lastModified = 13572468L;
+        entry.ttl = 9876543L;
+        entry.softTtl = 8765432L;
+        entry.etag = "etag";
+        entry.responseHeaders = new HashMap<>();
+        entry.responseHeaders.put("fruit", "banana");
+        entry.responseHeaders.put("color", "yellow");
+
+        putEntry("my-magical-key", entry).get();
+
+        assertThatEntriesAreEqual(getEntry("my-magical-key").get(), entry);
+        assertThat(getEntry("unknown-key").get(), is(nullValue()));
```

Comment:
can you do: 

```java
assertNull(getEntry(...).get());
```
?

Reply:
Yeah, I kept the same format from DiskBasedCacheTest. But I think doing assertNull makes more sense.

Reply:
Went through and changed this in DiskBasedCacheTest and DiskBasedCacheAsyncTest